libraries/chain/asset_evaluator.cpp

```diff
@@ -378,54 +378,60 @@ void check_children_of_bitasset(database& d, const asset_update_bitasset_operati
       return;
 
    // loop through all assets that have this asset as a backing asset
-   const auto& idx = d.get_index_type<asset_index>().indices().get<by_type>();
-
-   for( auto itr = idx.lower_bound(true); itr != idx.end(); ++itr )
+   const auto& idx = d.get_index_type<graphene::chain::asset_bitasset_data_index>().indices().get<by_short_backing_asset>();
+   auto backed_by_itr = idx.find( op.asset_to_update );
+   if ( backed_by_itr != idx.end() ) // if we found at least 1
    {
-      const auto& child = *itr;
-      if ( child.bitasset_data(d).options.short_backing_asset == op.asset_to_update )
+      auto backed_end = idx.upper_bound( new_backing_asset.get_id() );
+
+      for( ; backed_by_itr != backed_end; ++backed_by_itr )
       {
-         if ( after_hf_922_931 )
+         const auto& bitasset_data = *backed_by_itr;
+         const auto& child = bitasset_data.asset_id(d);
+         if ( bitasset_data.options.short_backing_asset == op.asset_to_update )
```

Comment:
After fixed the above one, this `if` is not needed.

Reply:
Avoiding this ``if`` was the purpose for this change. Thanks for being thorough and setting me straight.